ee/apps/omnichannel-transcript/src/OmnichannelTranscript.ts

```diff
@@ -1,16 +1,272 @@
+							return { name: file.name, buffer: null };
+						}
+
+						const fileBuffer = await this.uploadService.getFileBuffer({ userId, file: uploadedFile });
+						return { name: file.name, buffer: fileBuffer };
+					}),
+				);
+
+				// When you send a file message, the things you type in the modal are not "msg", they're in "description" of the attachment
+				// So, we'll fetch the the msg, if empty, go for the first description on an attachment, if empty, empty string
+				const msg = message.msg || message.attachments.find((attachment) => attachment.description)?.description || '';
+				// Remove nulls from final array
+				return { _id: message._id, msg, u: message.u, files: files.filter(Boolean), ts: message.ts };
+			}),
+		);
+	}
+
+	async workOnPdf({ template, details }: { template: Templates; details: WorkDetailsWithSource }): Promise<void> {
+		this.log.log(`Processing transcript for room ${details.rid} by user ${details.userId} - Received from queue`);
+		if (this.maxNumberOfConcurrentJobs <= this.currentJobNumber) {
+			this.log.error(`Processing transcript for room ${details.rid} by user ${details.userId} - Too many concurrent jobs, queuing again`);
+			throw new Error('retry');
+		}
+		this.currentJobNumber++;
+
+		const room = await LivechatRooms.findOneById(details.rid);
+		if (!room) {
+			throw new Error('room-not-found');
+		}
+		const messages = await this.getMessagesFromRoom({ rid: room._id });
+
+		const data = {
+			visitor:
+				room.v && (await LivechatVisitors.findOneById(room.v._id, { projection: { _id: 1, name: 1, username: 1, visitorEmails: 1 } })),
+			agent: room.servedBy && (await Users.findOneAgentById(room.servedBy._id, { projection: { _id: 1, name: 1, username: 1 } })),
+			closedAt: room.closedAt,
+			messages: await this.getFiles(details.userId, messages),
+		};
+
+		try {
+			await this.doRender({ template, data, details });
+		} catch (error) {
+			await this.pdfFailed({ details, e: error as Error });
+		}
+	}
+
+	async doRender({ template, data, details }: { template: Templates; data: WorkerData; details: WorkDetailsWithSource }): Promise<void> {
+		const buf: Uint8Array[] = [];
+		let outBuff = Buffer.alloc(0);
+
+		const stream = await this.worker.renderToStream({ template, data });
+		stream.on('data', (chunk) => {
+			buf.push(chunk);
+		});
+		stream.on('end', () => {
+			outBuff = Buffer.concat(buf);
+
+			return this.uploadService
+				.uploadFile({
+					userId: details.userId,
+					buffer: outBuff,
+					details: {
+						name: 'transcript.pdf',
+						type: 'application/pdf',
+						rid: details.rid,
+						// Rocket.cat is the goat
+						userId: 'rocket.cat',
+						size: outBuff.length,
+					},
+				})
+				.then((file) => this.pdfComplete({ details, file }))
+				.catch((e) => this.pdfFailed({ details, e }));
+		});
+	}
+
+	private async pdfFailed({ details, e }: { details: WorkDetailsWithSource; e: Error }): Promise<void> {
+		this.currentJobNumber--;
```

Comment:
```suggestion
```

Won't be needed if you decide to go this route -> https://github.com/RocketChat/Rocket.Chat/pull/27568/files#r1057216219